arangod/RocksDBEngine/RocksDBMetaCollection.cpp

```diff
@@ -1294,7 +1314,11 @@ Result RocksDBMetaCollection::bufferTruncate(rocksdb::SequenceNumber seq) {
     if (_revisionTreeApplied >= seq) {
       return;
     }
-    _revisionTruncateBuffer.emplace(seq);
+    auto [it, inserted] = _revisionTruncateBuffer.emplace(seq);
+    TRI_ASSERT(inserted);
+    if (inserted) {
```

Comment:
I prefer to do it only inside the `if`, because otherwise the logic would be incorrect if the entry wasn't inserted in non-maintainer mode.
It should not make a relevant difference in terms of performance if we the `if` here or not, but for correctness it can matter.